### DISCUSSION TITLE
How to get wire:key value inside php class component?

### DISCUSSION THREAD

**Discussion by massimomarazzi** [livewire/livewire#5551] [1. Help]
Hi there, 

I need to get wire:key value inside class component when updated hook run.

I try to define a property $key and next i read property into updated method but don't work.

Suggestion?

Thank you in advance

**Comment by joshhanley** [comment]
@massimomarazzi can you share the code of your component class and view showing what you're trying to do?

**Comment by massimomarazzi** [comment]
This component are nested into clonable component, i use repeaterIndex attribute but it is redundant because i have wire:key.

```
 <livewire:cities-autocomplete
                        wire:key="city_id.{{$index}}"
                        repeaterIndex="{{$index}}"
                        name="city_id.{{$index}}"
                        value="{{ $city_id }}"
                />
```

Into cities-autocomplete i get repeaterIndex, because i don't know how to get wire:key value.

```
$this->emit('selectedCity', [
                    'city_id' => $this->savedCityId,
                    'repeaterIndex' => $this->repeaterIndex
                ]
```

**Comment by joshhanley** [comment]
@massimomarazzi ah right, yeah you can't access the `wire:key` that I know of. As your key looks the same as the name, why don't you make use of the name field to dispatch with your event? That way you don't need the redundant `repeaterIndex`?
